packages/components/storybook/config.js

```diff
@@ -3,13 +3,15 @@
  */
 import { addDecorator, configure } from '@storybook/react';
 import { withA11y } from '@storybook/addon-a11y';
+import { withKnobs } from '@storybook/addon-knobs';
 
 /**
  * Internal dependencies
  */
 import '../build-style/style.css';
 
 addDecorator( withA11y );
+addDecorator( withKnobs );
```

Comment:
We should use story-level decorators to avoid having empty knob tabs in stories that don't have any knobs.

https://storybook.js.org/docs/formats/component-story-format/#story-exports

Reply:
I've tried to do this, but the "Knobs" tab still shows on all. See c57e049
Any suggestions?

Also, we can't name the story `withKnobs` if we import it in from the addon

Reply:
>I've tried to do this, but the "Knobs" tab still shows on all. See c57e049
>Any suggestions?

What happens if you delete `import '@storybook/addon-knobs/register';`?

>Also, we can't name the story withKnobs if we import it in from the addon

It can be `_withKnobs` just like Default is `_default`.

Reply:
Of course, `_withKnobs` for the name works fine.

Removing the knobs from `addons.js` then it doesn't show at all. I'll keep trying a few variations, the documentation is for various formats so hard to see what they may want for CSF.

Reply:
It looks like that is just how it works. If anything, let's just add it globally for now and figure out how to hook into the framework to hide it ourselves later.

Reply:
I'm sure we can find some CSS and `{ display: none; }` works wonders ;-)

Reply:
Pushed up the change. :+1:

Reply:
Thanks :smile: